### PR TITLE
CAMEL-20789: Download binary content of Secret or Config Map in property

### PR DESCRIPTION
**Pull request by mcarlett**
# Description

Improvement on camel-kubernetes, it is possible to save a binay file from Secret or Config Map

# Target

- [x ] I checked that the commit is targeting the correct branch (note that Camel 3 uses `camel-3.x`, whereas Camel 4 uses the `main` branch)

# Tracking
- [ x] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).

# Apache Camel coding standards and style

- [x ] I checked that each commit in the pull request has a meaningful subject line and body.

- [ x] I have run `mvn clean install -DskipTests` locally and I have committed all auto-generated changes

### CODE REVIEW

**Review by davsclaus — COMMENTED** [review]
This also requires to update the documentation in
docs/user-manual/modules/ROOT/pages/using-propertyplaceholder.adoc

File: components/camel-kubernetes/src/main/java/org/apache/camel/component/kubernetes/properties/SecretPropertiesFunction.java [hunk 1]
```diff
@@ -55,7 +55,13 @@ String lookup(String name, String key, String defaultValue) {
                 if (answer != null) {
                     byte[] data = Base64.getDecoder().decode(answer);
                     if (data != null) {
-                        answer = new String(data);
+                        if (binaryContent) {
+                            if (binaryContent) {
```
[the remark this answers is not in the document]
Comment (mcarlett) [reply components/camel-kubernetes/src/main/java/org/apache/camel/component/kubernetes/properties/SecretPropertiesFunction.java]: thanks @davsclaus

**Comment by mcarlett** [comment]
> This also requires to update the documentation in docs/user-manual/modules/ROOT/pages/using-propertyplaceholder.adoc

@davsclaus I noticed that `?` it is already used as optional property placeholder, I think I need to change the character used, any tip? thanks

**Comment by davsclaus** [comment]
You can maybe make new functions named `configmap-binary` and `secrets-binary` and then leave the existing as-is. Then its separated and its easy to spot in the name that they are binary or not.

**Comment by mcarlett** [comment]
> You can maybe make new functions named `configmap-binary` and `secrets-binary` and then leave the existing as-is. Then its separated and its easy to spot in the name that they are binary or not.

added `configmap-binary`  and  `secret-binary`, refactoring the existing code and adding new tests

**Comment by mcarlett** [comment]
doc updated

**Review by davsclaus — APPROVED** [review]

**Review by oscerd — APPROVED** [review]

**Pull request merged**